src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,432 @@
+import org.broadinstitute.barclay.argparser.CommandLinePluginDescriptor;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.walkers.annotator.Annotation;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.reflections.ReflectionUtils;
+
+import java.lang.reflect.Modifier;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A plugin descriptor for managing the dynamic discovery of both @{InfoFieldAnnotation} and @{GenotypeAnnotation} objects
+ * within the @{org.broadinstitute.hellbender.tools.walkers.annotator} package which handles integrating annotation specific
+ * arguments from the command line with tool specified defaults.
+ *
+ * Unlike @{GATKReadFilterPluginDescriptor} annotation order is not important and thus argument order is not guarinteed to be
+ * preserved in all cases, especially when group annotations are involved.
+ */
+public class GATKAnnotationPluginDescriptor  extends CommandLinePluginDescriptor<Annotation> {
+    private static final String pluginPackageName = "org.broadinstitute.hellbender.tools.walkers.annotator";
+    private static final Class<?> pluginBaseClass = org.broadinstitute.hellbender.tools.walkers.annotator.Annotation.class;
+
+    protected transient Logger logger = LogManager.getLogger(this.getClass());
+
+    @VisibleForTesting
+    @ArgumentCollection
+    private final VariantAnnotationArgumentCollection userArgs;
+
+    // Map of Annotation (simple) class names to the corresponding discovered plugin instance
+    private final Map<String, Annotation> allDiscoveredAnnotations = new HashMap<>();
+
+    // Map of Annotation (simple) class names to the corresponding default plugin instance
+    // it is a LinkedHashMap because we want to remember the order in which these were provided, and also keep the
+    // actual instances in case they have any additional state provided by the tool
+    // when they were created
+    private final Map<String, Annotation> toolDefaultAnnotations = new LinkedHashMap<>();
+    private final Set<String> toolDefaultGroups = new HashSet<>();
+
+    // Set of dependent args for which we've seen values (requires predecessor)
+    private final Set<String> requiredPredecessors = new HashSet<>();
+
+    private final Map<String, List<Annotation>> discoveredGroups = new HashMap<>();
+
+    /**
+     * @param userArgs           Argument collection to control the exposure of the command line arguments.
+     * @param toolDefaultFilters Default filters that may be supplied with arguments
+     *                           on the command line. May be null.
+     */
+    public GATKAnnotationPluginDescriptor(final VariantAnnotationArgumentCollection userArgs, final List<Annotation> toolDefaultFilters, final List<String> toolDefaultGroups) {
```

Comment:
filter

Reply:
I would suggest making toolDefaultGroups something more strongly typed than `List<String>`, like `List<Class<? extends Annotation>>`, or whatever is appropriate. I would also suggest validating them here and throwing if they're invalid, rather than waiting until validateArgs is called.

Reply:
done